### PR TITLE
Implement resource pages and links from site footer

### LINKED ISSUE #19: Implement resource pages and links from site footer
**Depends on #18**
*NOTE:* This issue only depends on #18 for implementing the actual links. The models can be implemented before that merges.  

Design: https://projects.invisionapp.com/share/KH11D3ZN53GC#/screens/455535008 (featured on all pages)

Each link at the bottom right of the footer corresponds to a resource page. We implemented these same pages in mitxpro, and we can basically copy that implementation over to this project directly.

**Acceptance criteria:**
- All relevant resource page models and relevant styles copied over from mitxpro and made editable in the CMS 
- Footer links to resource pages work as expected

### PR DESCRIPTION
**Pull request by umarmughal824**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
fixes #19 

#### What's this PR do?
Implement resource pages and links from site footer

#### How should this be manually tested?
Run the migrate command and then configure_wagtail that should populate three page Resource pages (About Us / Terms of Service / Honor Code / Privacy Policy)

#### Screenshots (if appropriate)
Screenshot 2021-07-27 at 13 41 28
Screenshot 2021-07-27 at 13 41 44
Screenshot 2021-07-27 at 14 05 03
Screenshot 2021-07-27 at 17 13 25
Screenshot 2021-07-27 at 17 13 40
Screenshot 2021-07-27 at 17 14 09
Screenshot 2021-07-27 at 17 14 20
Screenshot 2021-07-27 at 17 14 29
Screenshot 2021-07-27 at 17 27 06
Screenshot 2021-07-27 at 17 27 30
Screenshot 2021-07-27 at 17 27 39
Screenshot 2021-07-27 at 17 37 58
Screenshot 2021-07-27 at 17 38 06

### CODE REVIEW

**Comment by umarmughal824** [comment]
@pdpinch could you please review the attached screenshots in the PR description.

**Comment by pdpinch** [comment]
Why are there double-colons `::` in the mobile screenshots? 

Everything else seems fine. Please have someone from Arbisoft / Edly do a code review.

**Comment by umarmughal824** [comment]
> Why are there double-colons `::` in the mobile screenshots?

@pdpinch that's because I have added extra `:` in the content on the wagtail page title section.

I will get it reviewed once I will be done with the test cases.

**Comment by pdpinch** [comment]
Can you add some `help_text` for the CMS user with the ideal image dimensions? on xPRO we found that CMS users were uploading much larger images than they needed to, which was bringing down our Lighthouse score.

File: cms/api.py [hunk 1]
```diff
@@ -13,6 +15,31 @@
 DEFAULT_SITE_PROPS = dict(hostname="localhost", port=80)
 
 
+def create_resource_page_under_parent(title, parent):
```
Comment (gsidebo) [inline cms/api.py:18]:
```suggestion
def get_or_create_resource_page(title, parent):
```

File: cms/api.py [hunk 2]
```diff
@@ -25,6 +52,20 @@ def get_home_page():
...
+    Ensures that all the following resource pages exists
+    (About Us / Terms of Service / Privacy Policy / Honor Code)
+    """
+    home_page = Page.objects.filter(
+        content_type=ContentType.objects.get_for_model(cms_models.HomePage)
+    ).first()
+    create_resource_page_under_parent("About Us", home_page)
+    create_resource_page_under_parent("Terms of Service", home_page)
+    create_resource_page_under_parent("Privacy Policy", home_page)
+    create_resource_page_under_parent("Honor Code", home_page)
```
Comment (gsidebo) [inline cms/api.py:66]: Kwargs would make these a little clearer:

`get_or_create_resource_page("About Us", parent=home_page)`

File: cms/models.py [hunk 3]
```diff
@@ -71,3 +77,38 @@ class CoursePage(ProductPage):
     )
 
     content_panels = [FieldPanel("course")] + ProductPage.content_panels
+
+
+class ResourcePage(Page):
+    """
+    Basic resource page for all resource page.
```
Comment (gsidebo) [inline cms/models.py:84]:
```suggestion
    Basic resource page class for pages containing basic information (FAQ, etc.)
```

**Review by gsidebo — CHANGES_REQUESTED** [review]

**Review by gsidebo — APPROVED** [review]
There are some ways that we should update this to match the designs more closely, but we have a bunch of open PRs and it's much better to merge this as it is now, then create a separate issue for the updates. LGTM! 👍

**Comment by umarmughal824** [comment]
> There are some ways that we should update this to match the designs more closely, but we have a bunch of open PRs and it's much better to merge this as it is now, then create a separate issue for the updates. LGTM! 👍

@gsidebo As I could not see the design for resource pages on invision so I can't say anything about it. If you have anything in mind that requires updates so please create an issue for that with the details.

**Comment by briangrossman** [comment]
Tested it out on RC. Looks good! 👍 

- [About Us](https://mitxonline-rc.mitxonline.mit.edu/about-us/)
- [Terms of Service](https://mitxonline-rc.mitxonline.mit.edu/terms-of-service/)
- [Privacy Policy](https://mitxonline-rc.mitxonline.mit.edu/privacy-policy/)
- [Honor Code](https://mitxonline-rc.mitxonline.mit.edu/honor-code/)

Checking the box on the release...

**Comment by briangrossman** [comment]
FYI: Updated these pages to include the content from xPRO. 

I changed every `MIT xPRO` => `MITx Online`. They will require a review to check for content problems. In particular, links and emails should be checked / updated. 

Of note:, xPRO has a `Subheading` at the top of each of these pages
image

... but MITx Online has a `Header Image` instead.
image

**Pull request merged**